packages/integrations/vercel/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import type { AstroIntegration, AstroConfig } from 'astro';
+import fs from 'fs/promises';
+import type { PathLike } from 'fs';
+
+export type { VercelApiHandler, VercelRequest, VercelRequestBody, VercelRequestCookies, VercelRequestQuery, VercelResponse } from '@vercel/node';
+
+const writeJson = (path: PathLike, data: any) => fs.writeFile(path, JSON.stringify(data), { encoding: 'utf-8' });
+
+export function vercelFunctions(): AstroIntegration {
+	let _config: AstroConfig;
+	let output: URL;
+	return {
+		name: '@astrojs/vercel',
+		hooks: {
+			'astro:config:setup': ({ config }) => {
+				output = new URL('./.output/', config.projectRoot);
+				config.dist = new URL('./static/', output);
+				config.buildOptions.pageUrlFormat = 'directory';
+			},
+			'astro:config:done': async ({ config, setAdapter }) => {
+				// setAdapter(getAdapter(config.buildOptions.site));
```

Comment:
This is not really an adapter since it's not calling `setAdapter`. We should probably be throwing if `adapter: something()` is set in the config but that adapter does not call `setAdapter`.

Since it looks like you are wanting to SSG for now, can you look at the default adapter which just sets one config flag to turn on SSG: https://github.com/withastro/astro/blob/main/packages/astro/src/adapter-ssg/index.ts#L19

Reply:
@matthewp +1 for throwing if adapter doesn't call setAdapter.